### PR TITLE
Add watch sync/async watch functions to store client

### LINKED ISSUE #10: Listen to S3 bucket notifications via Minio Python API
See: https://min.io/docs/minio/linux/administration/monitoring/publish-events-to-mqtt.html

We need this to implement the Python helper function (#7) which watches for added/deleted/modified objects in the datastore.

### PR DESCRIPTION
**Pull request by stv0g**
Fixes #10

### CODE REVIEW

File: seguro/common/store.py [hunk 1]
```diff
@@ -54,89 +83,88 @@ def __init__(
...
+        return Watcher(self, prefix, cb, events)
 
-        Arguments:
-            filename -- Local filename that is used
 
-        Keyword arguments:
-            bucket -- Storage bucket that is considered (default "seguro")
+class Watcher(threading.Thread):
+    def __init__(
+        self, client: Client, prefix: str, cb: callable, events: EventType
```
Comment (fwege) [inline seguro/common/store.py:151]: Do I understand correctly that with that implementation we need individual `watcher` instances for each folder, i.e. prefix, we want to watch? Is that reasonable?

File: seguro/common/test_store.py [hunk 2]
```diff
@@ -21,27 +21,73 @@ def test_store():
...
+            raise StopIteration()
+
+        i += 1
+
+    filename = "some/prefix/test"
+
+    store.remove_file(filename)
+    watcher = store.watch_async("some/prefix/", callback)
+
+    watcher.start()
```
Comment (fwege) [inline seguro/common/test_store.py:88]: Is there a reason to force the user call `watcher.start()` manually after defining which prefix to watch?
Couldn't that be simplified by starting the watcher as soon as a prefix is registered for async watching, and stopping it when the watcher object is destroyed?

**Review by fwege — COMMENTED** [review]
Nice work! I added two questions to make sure I understand your implementation correctly :)

File: seguro/common/store.py (hunk 1 above)
Comment (stv0g) [reply seguro/common/store.py:151]: Yes, unfortunately thats the case as the minio Python package does not support asynchronous IO natively.
In its core its doing a blocking read on the HTTP socket. Hence we must create a new watcher (OS thread) per prefix.

If we want to get fancy, we should use an async IO compatible minio package like this one: https://github.com/hlf20010508/miniopy-async

File: seguro/common/test_store.py (hunk 2 above)
Comment (stv0g) [reply seguro/common/test_store.py:88]: Yeah, thats a good point. We should start the watcher right away :)

Comment (stv0g) [reply seguro/common/test_store.py:88]: I was following the sematics of `threading.Thread` which also needs to be started manually

File: seguro/common/store.py (hunk 1 above)
Comment (fwege) [reply seguro/common/store.py:151]: Ok, but I think that is okay for the moment as we can still watch an entire prefix. When defining the directory hierarchy of the store, we should keep that mind as it influences how convenient we can track files.

File: seguro/common/test_store.py (hunk 2 above)
Comment (fwege) [reply seguro/common/test_store.py:88]: I see. If possible I would still go with starting it right away... :)

File: seguro/common/store.py (hunk 1 above)
Comment (stv0g) [reply seguro/common/store.py:151]: Minio also supports watching for a prefix + suffix match.
However, I didnt expose this option via the API yet

File: seguro/common/test_store.py (hunk 2 above)
Comment (stv0g) [reply seguro/common/test_store.py]: Addressed in 8404727468f01b3227682f0363ad6637ab1810c8

**Review by fwege — APPROVED** [review]

**Pull request merged**